### PR TITLE
MAP-1383 rework cell capacity search to use locations api

### PR DESCRIPTION
**Pull request by Mjwillis**
This rather large change does the following:
- uses the location service to find cells with capacity
- removes the need for a second lookup of prisoner in locations as above returns all needed data
- switches to use the correct non-association endpoint which automatically filters out expired, and prisoners not in the current prison
- starts to move a lot of calls to prison-api to get prisoner data from prisoner search - quick and removes need to lookup additional information.

### CODE REVIEW

File: server/controllers/cellMove/cellMoveUtils.test.ts [hunk 1]
```diff
@@ -1,260 +1,96 @@
...
-          offenderNonAssociation: expect.objectContaining({
+          otherPrisonerDetails: expect.objectContaining({
             firstName: 'Clarence',
             lastName: 'Cook',
           }),
         }),
       )
     })
-
-    it('filters out expired non-associations', async () => {
```
Comment (Mjwillis) [inline server/controllers/cellMove/cellMoveUtils.test.ts:215]: The API does all this now

File: server/views/receptionMove/partials/nonAssociations.njk [hunk 2]
```diff
@@ -25,6 +25,14 @@
               text: nonAssociation.prisonNumber
             }
           },
+          {
+            key: {
+              text: "Reason"
```
Comment (Mjwillis) [inline server/views/receptionMove/partials/nonAssociations.njk:30]: Add the very useful reason field

File: server/services/prisonerDetailsService.ts [hunk 3]
```diff
@@ -11,6 +15,20 @@ export default class PrisonerDetailsService {
     return this.prisonApiClient.getPrisonerImage(token, offenderNo, fullSizeImage)
   }
 
+  async getPrisoner(token: string, prisonerNumber: string): Promise<Prisoner> {
```
Comment (Mjwillis) [inline server/services/prisonerDetailsService.ts:18]: Started to use prisoner search for lookups

File: server/data/whereaboutsApiClient.test.ts [hunk 4]
```diff
@@ -21,20 +21,6 @@ describe('whereaboutsApiClient', () => {
     nock.cleanAll()
   })
 
-  describe('getCellsWithCapacity', () => {
```
Comment (Mjwillis) [inline server/data/whereaboutsApiClient.test.ts:24]: Removed

File: server/data/locationsInsidePrisonApiClient.ts [hunk 5]
```diff
@@ -62,4 +83,12 @@ export default class LocationsInsidePrisonApiClient {
       path: `/locations/prison/${agencyId}/group/${groupName}/location-prefix`,
     })
   }
+
+  getCellsWithCapacity(token: string, agencyId: string, groupName: string = null): Promise<CellLocation[]> {
```
Comment (Mjwillis) [inline server/data/locationsInsidePrisonApiClient.ts:87]: This is the core change

File: server/data/nonAssociationsApiClient.ts [hunk 6]
```diff
@@ -37,9 +45,9 @@ export default class NonAssociationsApiClient {
     return new RestClient('Non-associations Api Client', config.apis.nonAssociationsApi, token)
   }
 
-  getNonAssociationsLegacy(token: string, offenderNo: string): Promise<OffenderNonAssociationLegacy> {
-    return NonAssociationsApiClient.restClient(token).get<OffenderNonAssociationLegacy>({
-      path: `/legacy/api/offenders/${offenderNo}/non-association-details`,
+  getNonAssociations(token: string, prisonerNumber: string): Promise<PrisonerNonAssociation> {
+    return NonAssociationsApiClient.restClient(token).get<PrisonerNonAssociation>({
+      path: `/prisoner/${prisonerNumber}/non-associations?includeOpen=true&includeClosed=false&includeOtherPrisons=false&sortBy=WHEN_UPDATED&sortDirection=DESC`,
```
Comment (Mjwillis) [inline server/data/nonAssociationsApiClient.ts:50]: The API does the hard work now

File: server/controllers/cellMove/selectCell.ts [hunk 7]
```diff
@@ -1,117 +1,78 @@
...
-import { OffenderCell } from '../../data/prisonApiClient'
+import { Prisoner } from '../../data/prisonerSearchApiClient'
 
 const defaultSubLocationsValue = { text: 'Select area in residential unit', value: '' }
 const noAreasSelectedDropDownValue = { text: 'No areas to select', value: '' }
 const toDropDownValue = entry => ({ text: entry.name, value: entry.key })
 
-const sortByDescription = (a, b) => a.description.localeCompare(b.description)
-
-const getCellOccupants = async (
```
Comment (Mjwillis) [inline server/controllers/cellMove/selectCell.ts:26]: None of this needed! All taken care of by the APIs

File: integration_tests/mockApis/responses/prisonerBasicDetails.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "prisonerNumber": "ABC123",
```
Comment (steverendell) [inline integration_tests/mockApis/responses/prisonerBasicDetails.json:2]: A12345 crops up a fair bit too - I suppose these should have the right format or could confuse people in future

**Review by steverendell — APPROVED** [review]
lgtm

File: server/data/nonAssociationsApiClient.ts (hunk 6 above)
Comment (GurnankCheema) [reply server/data/nonAssociationsApiClient.ts:50]: nice 👍

File: server/controllers/cellMove/confirmCellMove.ts [hunk 9]
```diff
@@ -49,7 +54,7 @@ export default ({
       ? { pathHierarchy: 'swap' }
       : await locationService.getLocation(systemClientToken, cellId)
 
-    const { firstName, lastName } = await prisonerDetailsService.getDetails(systemClientToken, offenderNo)
+    const { firstName, lastName } = await prisonerDetailsService.getPrisoner(systemClientToken, offenderNo)
```
Comment (GurnankCheema) [inline server/controllers/cellMove/confirmCellMove.ts:57]: nice

**Pull request merged**